src/utils/dynamic-import-plugin.js

```diff
@@ -0,0 +1,42 @@
+// Dynamic Import support for acorn
+export default function wrapDynamicImportPlugin ( acorn ) {
+  let moduleDynamicImportsReturnBinding;
+  acorn.tokTypes._import.startsExpr = true;
+  acorn.plugins.dynamicImport = ( instance ) => {
+    instance.extend( 'parseStatement', nextMethod => {
+      return function parseStatement ( ...args ) {
+        const node = this.startNode();
+        if ( this.type === acorn.tokTypes._import ) {
+          const nextToken = this.input[this.pos];
+          if ( nextToken === acorn.tokTypes.parenL.label ) {
+            const expr = this.parseExpression();
+            return this.parseExpressionStatement( node, expr );
+          }
+        }
+        return nextMethod.apply( this, args );
+      }
+    });
+
+    instance.extend( 'parseExprAtom', nextMethod => {
+      return function parseExprAtom ( refDestructuringErrors ) {
+        if ( this.type === acorn.tokTypes._import ) {
+          const node = this.startNode();
+          this.next();
+          if ( this.type !== acorn.tokTypes.parenL ) {
+            this.unexpected();
+          }
+          if ( moduleDynamicImportsReturnBinding ) {
+            moduleDynamicImportsReturnBinding.push( node );
+          }
+          return this.finishNode( node, 'Import' );
+        }
+        return nextMethod.call( this, refDestructuringErrors );
+      };
+    });
+  };
+
+  // returns a function to set the dynamicImport array for getting these nodes during parsing
+  return function setModuleDynamicImportsReturnBinding ( _moduleDynamicImportsReturnBinding ) {
+    moduleDynamicImportsReturnBinding = _moduleDynamicImportsReturnBinding;
+  };
+}
```

Comment:
I'd still prefer this to use the existing plugin, but I can understand that you don't want to have a dependency on another different acorn version. I hope kesne/acorn-dynamic-import#11 gets merged soon, otherwise I'll fork the repo.

Reply:
I agree that having a publicly supported plugin is probably preferable as long as it is properly supported. However since now everything is behind a feature flag and the plugin logic is not exposed to the user, changing this later should not be a problem.

Reply:
Sure, this is fine as a start. Since I'm using `acorn-dynamic-import` in [acorn-stage3](https://github.com/adrianheine/acorn-stage3), I want to make sure one way or the other that there is a maintained version.

Reply:
Let me know if you end up forking it and I can change my pull request at https://github.com/kesne/acorn-dynamic-import/pull/10 to point to the new fork.

Reply:
kesne/acorn-dynamic-import#11 is merged and there should be a release soon (after kesne/acorn-dynamic-import#10 landed, too). Should I prepare a PR?

Reply:
@adrianheine if you're interested that would be amazing. Would likely have to come after https://github.com/rollup/rollup/pull/1857.

Reply:
See #1891.